File: MoonApp/packages/react-native-nitro-moon/ios/Common/CreateError.swift
import Foundation

func CreateError(code: Int = 400, message: String) -> NSError {
    return NSError(
        domain: Bundle.main.bundleIdentifier ?? "Error",
        code: code,
        userInfo: [
            NSLocalizedDescriptionKey: message
        ]
    )
}

extension NSError {
    func toJSONString() -> String {
        let dict: [String: Any] = [
            "domain": self.domain,
            "code": self.code,
            "message": self.localizedDescription
        ]
        
        if let data = try? JSONSerialization.data(withJSONObject: dict, options: []) {
            return String(data: data, encoding: .utf8) ?? "{}"
        } else {
            return "{}"
        }
    }
}

File: MoonAdmin/src/composables/CreateError.js

export class CreateError extends Error {
    constructor(code, message) {
        super(message)
        this.code = code || 502
        Error.captureStackTrace(this, CreateError);
    }

    get statusCode() {
        return this.code
    }

    toString() {
        return this.message
    }

    toJSON() {
        return {
            code: this.code,
            message: this.message
        }
    }

}